notifier/notifier.go

```diff
@@ -472,17 +473,30 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 	)
 	for _, ams := range amSets {
 		var (
-			payload []byte
-			err     error
+			payload  []byte
+			err      error
+			amAlerts = alerts
 		)
 
 		ams.mtx.RLock()
 
+		if len(ams.cfg.AlertRelabelConfigs) > 0 {
+			amAlerts = relabelAlerts(ams.cfg.AlertRelabelConfigs, labels.Labels{}, alerts)
+			// TODO(nabokihms): figure out the right way to cache marshalled alerts.
```

Comment:
What do you mean here?

Reply:
`v1Payload` and `v2Payload` were previously used to keep the result of marshaling alerts. The case when two alert managers used the same protocol version was optimized. It is not optimized, but if users start using the new per alert manager relabeling, it may be less performant.

Reply:
It is a bug to re-use the marshalled version for an alertmanager with no relabeling, after one that has some.

Reply:
@bboreham do you want me to fix this? This was an attempt to consume less resources as far as I remember. I can open the following PR with my variant of the fix.

Reply:
If you can fix the bug, please do.

Reply:
I fixed it #13735